core/src/block.rs

```diff
@@ -684,27 +675,32 @@ mod valid {
         }
     }
 
-    // See also [SignedBlockCandidate::validate_genesis]
-    fn check_genesis_block(
-        block: &SignedBlock,
-        genesis_account: &AccountId,
-    ) -> Result<(), InvalidGenesisError> {
-        let signatures = block.signatures().collect::<Vec<_>>();
-        let [signature] = signatures.as_slice() else {
-            return Err(InvalidGenesisError::InvalidSignature);
-        };
-        signature
-            .1
-            .verify(&genesis_account.signatory, block.payload())
-            .map_err(|_| InvalidGenesisError::InvalidSignature)?;
-
-        let transactions = block.payload().transactions.as_slice();
-        for transaction in transactions {
-            if transaction.value.authority() != genesis_account {
-                return Err(InvalidGenesisError::UnexpectedAuthority);
-            }
-        }
-        Ok(())
+    fn add_genesis_domain_and_account(state_block: &mut StateBlock, genesis_block: &SignedBlock) {
+        let genesis_account = genesis_block
+            .transactions()
+            .next()
+            .expect("Genesis block has transactions")
+            .value
+            .authority();
+        let genesis_public_key = genesis_account.signatory.clone();
+        let (genesis_account, genesis_domain) = genesis_account_and_domain(genesis_public_key);
+        state_block
+            .world
+            .domains
+            .insert(genesis_domain.id.clone(), genesis_domain);
+        state_block
+            .world
+            .accounts
+            .insert(genesis_account.id.clone(), genesis_account);
+    }
+
+    fn genesis_account_and_domain(public_key: PublicKey) -> (Account, Domain) {
+        let genesis_account_id =
+            AccountId::new(iroha_genesis::GENESIS_DOMAIN_ID.clone(), public_key);
+        let genesis_account = Account::new(genesis_account_id).into_account();
+        let genesis_domain =
+            Domain::new(iroha_genesis::GENESIS_DOMAIN_ID.clone()).build(&genesis_account.id);
+        (genesis_account, genesis_domain)
```

Comment:
```suggestion
    fn genesis_account_and_domain(public_key: PublicKey) -> (Account, Domain) {
        let genesis_account_id =
            AccountId::new(iroha_genesis::GENESIS_DOMAIN_ID.clone(), public_key);
        let genesis_account = Account::new(genesis_account_id).into_account();
        let genesis_domain =
            Domain::new(iroha_genesis::GENESIS_DOMAIN_ID.clone()).build(&genesis_account.id);
        (genesis_account, genesis_domain)
```

IMO we should take both from the genesis authority. And remove the `GENESIS_DOMAIN_ID`